File: python/TapeEquilibrium.py
def solution(A):
    if(len(A)==0 or len(A)==1):
        return 0

    rhs = 0
    lhs = 0

    for a in A:
        rhs += a

    rhs -= A[0]
    lhs += A[0]
    min_diff = abs(lhs - rhs)
    
    for i in range(2, len(A)):
        lhs += A[i-1]
        rhs -= A[i-1]
        diff = abs(lhs - rhs)
        min_diff = min(min_diff, diff)

    return min_diff

File: cpp/TapeEquilibrium.cpp
#include <limits.h>
#include <math.h>

int solution(vector<int> &A) {
    // Implement your solution here
    int first = 0;
    int second = 0;
    int min_diff = INT_MAX;
    for(auto &a : A)
        second += a;

    for(unsigned i = 1; i<A.size(); i++)
    {
        first += A[i-1];
        second -= A[i-1];
        int diff = abs(first - second);
        if (diff < min_diff) 
            min_diff = diff;
    }

    return min_diff;
}
